### PR TITLE
Add Feast overview to README

### PR DESCRIPTION
**Pull request by woop**
- Added a short overview of Feast in README.md
- Added components.md to /docs
- Added concepts.md to /docs

### CODE REVIEW

File: docs/concepts.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+A Feature is an individual measurable property or characteristic of an Entity. In the context of Feast a Feature has the following attributes: 
+
+* Entity - It must be associated with a known Entity within Feast
+* ValueType - The feature type must be defined, e.g. String, Bytes, Int64, Int32, Float etc.
+* Requirements - Properties related to how a feature should be stored for serving and training
+* Granularity - Time series features require a defined granularity
+* StorageType - For both serving and training a storage type must be defined
+
+Feast needs to know these attributes for features in order to be able to ingest, store and serve a feature. Importantly a Feature is only a feature when Feast knows about it; This seems contrite, but in essence it introduces a best practice whereby a feature becomes available for ingestion, serving and training in production only when Feast has added it to it’s feature catalog and made it available.
```
Comment (zhilingc) [inline docs/concepts.md]:
```suggestion
Feast needs to know these attributes in order to be able to ingest, store and serve a feature. A Feature is only a feature when Feast knows about it; This seems contrite, but it introduces a best practice whereby a feature only becomes available for ingestion, serving and training in production when Feast has added the feature to its catalog.
```

File: docs/components.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+### Client Libraries
+
+Feast provides multiple client libraries for interacting with a Feast deployment.
+    
+| Functionality                | CLI | Go  | Java | Python (WIP)|
+|------------------------------|-----|-----|------|-------------|
+| Feature Management           | yes | no  | no   | yes         |
+| Data Ingestion (Jobs)        | yes | no  | no   | yes         |
+| Feature Retrieval (Training) | no  | no  | no   | yes         |
+| Feature Retreival (Serving)  | no  | yes | yes  | yes         |
```
Comment (zhilingc) [inline docs/components.md]: typo in `Retreival`

File: docs/components.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+
+# Components
+
+### Core
+
+Feast Core is the central component that manages Feast and all other components within the system. It allows for the registration and management of entities, features, data stores, and other system resources. Core also manages the execution of feature ingestion jobs from batch and streaming sources, and provides Feast Serving with feature related information for the retrieval of feature data.
```
Comment (zhilingc) [inline docs/components.md]:
```suggestion
Feast Core is the central component that manages Feast and all other components within the system. It allows for the registration and management of entities, features, data stores, and other system resources. Core also manages the execution of feature ingestion jobs from batch and streaming sources, and provides the other Feast components with feature related information.
```
It provides ingestion with the feature specs too

File: docs/concepts.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+### What is an Entity?
+
+An entity is a type with an associated key which generally maps onto a known domain object, e.g. Driver, Customer, Area, Merchant etc. An entity can also be a composite of other entities, with the corresponding composite key, e.g. DriverArea.
```
Comment (zhilingc) [inline docs/concepts.md]: ```An entity can also be a composite of other entities, with the corresponding composite key, e.g. DriverArea.```
Are we supporting this in the future?

File: docs/concepts.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+### What is Feast?
+Feast is a Feature Storage platform for Machine Learning features with the following  attributes:
+
+1. Ingestion and storage of ML features via batch or stream
+2. Retrieval of ML features for serving via API, or via Google BigQuery to create training datasets
+3. Maintaining of a feature catalog, including additional feature attribute information and discovery via API 
+
+Feast solves a need for standardising how features are stored, served and accessed, and encourages sharing and reuse of created features amongst data science teams.
+
+Feast does not prescribe how Features should be created but does allow for ingestion via batch or stream in a number of formats, e.g. batch import from CSV, BigQuery tables, streaming via Pub/Sub etc.
```
Comment (zhilingc) [inline docs/concepts.md]:
```suggestion
Feast does not prescribe how Features should be created. It allows for ingestion via batch or stream in a number of formats, e.g. batch import from CSV, BigQuery tables, streaming via Pub/Sub etc.
```

**Comment by woop** [comment]
I've squashed the commits after merging all suggestions @zhilingc @pradithya

**Pull request merged**